modular/uploader/upload_task.go

```diff
@@ -70,6 +74,19 @@ func (u *UploadModular) PreUploadObject(ctx context.Context, uploadObjectTask co
 		log.CtxErrorw(ctx, "failed to begin upload object task")
 		return err
 	}
+
+	taskState, _, errGetUploadObjectState := u.baseApp.GfSpClient().GetUploadObjectState(ctx, uploadObjectTask.GetObjectInfo().Id.Uint64())
+	if errGetUploadObjectState != nil {
+		log.CtxErrorw(ctx, "failed to get upload object state")
+		return ErrGetObjectUploadState
+	}
+	if taskState == int32(types.TaskState_TASK_STATE_UPLOAD_OBJECT_DONE) ||
+		taskState == int32(types.TaskState_TASK_STATE_REPLICATE_OBJECT_DOING) {
+		// It is not allowed to upload piece or object for an object id which had already been fully uploaded.
+		log.CtxErrorw(ctx, "failed to put object as the target object had already fully uploaded")
+		return ErrInvalidUploadRequest
+	}
```

Comment:
These codes can encapsulate a common check function, which is called by the upload and Resumable upload functions to facilitate unified modification; the state comparison does not seem to cover all and may cover more states.

Reply:
Fixed